src/main/java/com/arpnetworking/tsdcore/sinks/DimensionFilteringSink.java

```diff
@@ -33,6 +34,12 @@
      */
     @Override
     public void recordAggregateData(final PeriodicData data) {
+        if (_whitelistDimensions.isPresent()) {
+            if (!_whitelistDimensions.get().containsAll(data.getDimensions().keySet())) {
```

Comment:
Sorry, maybe I misunderstood where this was going. Are you intending to protect dimension award sinks like Wavefront or are you going to use this on "legacy" sinks liks telemetry?

I was guessing the latter where we want to only allow aggregations with some defined set, and all of that set with nothing more. 

Your second comment is more concerning, the image implementation in MAD has already been unified and the implementation here will be too. In the meantime your assertion may be correct, but I would validate that (that legacy dimensions are only on fqdsn). 

Also, are you aware that you will need to build this into MAD as well?